app/models/service_order.rb

```diff
@@ -73,9 +73,11 @@ def clear
 
   def self.add_to_cart(request, requester)
```

Comment:
@gmcculloug should `.add_to_cart` move to the `ServiceOrderCart` subclass or does v2v use this as well?

Reply:
I definitely see moving the `add_to_cart` and `remove_from_cart` operations into the new `ServiceOrderCart` model but I am a little concerned that it will require changes to the API repo where `remove_from_cart` is currently used: [app/controllers/api/subcollections/service_requests.rb - remove_service_request](https://github.com/ManageIQ/manageiq-api/blob/62de390490c8bc333f27feedfc58455ced1e7483/app/controllers/api/subcollections/service_requests.rb#L78)

If API changes are required I'd rather do that in a followup instead of tying another PR to this and the schema changes which would all need to be merged together.

@abellotti Can you provide some guidance on this?

Reply:
Yeah let's investigate in a follow-up

Reply:
The API code is used by SUI... (and only by SUI as far as I know):

add item:
`POST /api/service_orders/cart/service_requests`
```
action: 'add' 
resources: [ service_request ]
```

remove item:
`POST /api/service_orders/cart/service_requests`
```
action: 'remove' 
resources: [ { id: service_request.id } ]
```

order the cart:
`POST /api/service_orders/cart`
`action: 'order'`

clear the cart:
`POST /api/service_orders/cart`
`action: 'clear'`

list the cart:
`GET /api/service_orders/cart/service_requests?expand=resources`

So, these actions should only be valid on the special `cart` id, not on any other service_order instance.